libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/push/DefaultPushHandler.kt

```diff
@@ -75,21 +75,34 @@ class DefaultPushHandler @Inject constructor(
                 for (request in requests) {
                     // Log the result of the push notification event
                     val result = resolvedEvents[request]
-                    if (result?.isSuccess == true) {
-                        pushHistoryService.onSuccess(
+                    if (result == null) {
+                        pushHistoryService.onUnableToResolveEvent(
                             providerInfo = request.providerInfo,
                             eventId = request.eventId,
                             roomId = request.roomId,
                             sessionId = request.sessionId,
-                            comment = "Push handled successfully",
+                            reason = "Push not handled: no result found for request",
                         )
                     } else {
-                        pushHistoryService.onUnableToResolveEvent(
-                            providerInfo = request.providerInfo,
-                            eventId = request.eventId,
-                            roomId = request.roomId,
-                            sessionId = request.sessionId,
-                            reason = "Push not handled",
+                        result.fold(
+                            onSuccess = {
+                                pushHistoryService.onSuccess(
+                                    providerInfo = request.providerInfo,
+                                    eventId = request.eventId,
+                                    roomId = request.roomId,
+                                    sessionId = request.sessionId,
+                                    comment = "Push handled successfully",
+                                )
+                            },
+                            onFailure = { exception ->
+                                pushHistoryService.onUnableToResolveEvent(
+                                    providerInfo = request.providerInfo,
+                                    eventId = request.eventId,
+                                    roomId = request.roomId,
+                                    sessionId = request.sessionId,
+                                    reason = exception.message ?: exception.javaClass.simpleName,
```

Comment:
This is matching what we had [before](https://github.com/element-hq/element-x-android/pull/4722/files#diff-2d8c2336c93c4152ea5c38b36b8aae57056727b09313bc1e12f525635d71c0feL158), thanks!